app/backend/src/couchers/descriptor_pool.py

```diff
@@ -0,0 +1,21 @@
+import functools
+from pathlib import Path
+
+from google.protobuf import descriptor_pb2, descriptor_pool
+
+
+@functools.cache
+def get_descriptor_pool():
+    """
+    Generates a protocol buffer object descriptor pool which allows looking up info about our proto API, such as options
+    for each servicer, method, or message.
+    """
+    # this needs to be imported so the annotations are available in the generated pool...
+    from proto import annotations_pb2
+
+    pool = descriptor_pool.DescriptorPool()
+    with open(Path(__file__).parent / ".." / "protos.pb", "rb") as descriptor_set_f:
+        desc = descriptor_pb2.FileDescriptorSet.FromString(descriptor_set_f.read())
+    for file_descriptor in desc.file:
+        pool.Add(file_descriptor)
```

Comment:
I get a cryptic error when running test_api.py (without running any other test file first)
```
        pool = descriptor_pool.DescriptorPool()
        with open(Path(__file__).parent / ".." / "proto" / "descriptors.pb", "rb") as descriptor_set_f:
            desc = descriptor_pb2.FileDescriptorSet.FromString(descriptor_set_f.read())
        for file_descriptor in desc.file:
>           pool.Add(file_descriptor)
E           TypeError: Couldn't build proto file into descriptor pool!
E           Invalid proto descriptor for file "annotations.proto":
E             google/protobuf/descriptor.proto: Import "google/protobuf/descriptor.proto" has not been loaded.
E             sensitive: ".google.protobuf.FieldOptions" is not defined.
E             auth_level: ".google.protobuf.ServiceOptions" is not defined.

../couchers/descriptor_pool.py:20: TypeError
```

Reply:
That's indeed very cryptic.

Reply:
I'm not able to replicate this. Are you sure you generated the protos correctly, including with the `--include_imports` and `--include_source_info` flags?

Reply:
Thanks for testing! It was indeed the `--include_imports` that was missing from my homemade proto generator script. Resolving!